### PR TITLE
Add Show/Hide functionality + clean up failing AC.

### PR DESCRIPTION
**Pull request by mdbenjam**
The main purpose of this PR is to add the show/hide menu functionality. We are also cleaning up some neglected AC's from other tickets.

From #1261: Hook up the hide/open menu buttons to actually hide and open the menu.

From #1259: Change 'View all documents' to 'Back to all documents'. And make the doc type responsive to large document types.

From #1268: Change 'Jump to Section' to 'Jump to section'. And make the list view use a grid so that two digit page numbers don't move the start of the comment window.

From #1450: We flip the positioning of Edit and Delete. And use h4 instead of strong for the Page numbers to get the right text color.

From #1283: Update the spacing between the save/cancel buttons and the edit comment textbox. (The last AC).

### CODE REVIEW

File: client/app/components/PdfSidebar.jsx [hunk 1]
```diff
@@ -99,26 +99,26 @@ export default class PdfSidebar extends React.Component {
         </Comment>;
     });
 
-    return <div className="cf-sidebar-wrapper">
+    const sidebarClass = this.props.hidePdfSidebar ?
```
Comment (NickHeiner) [inline client/app/components/PdfSidebar.jsx]: Reminder to use [`classnames`](https://www.npmjs.com/package/classnames).

File: client/app/components/PdfSidebar.jsx [hunk 2]
```diff
@@ -164,6 +164,24 @@ export default class PdfSidebar extends React.Component {
...
 
+const mapSidebarDispatchToProps = (dispatch) => ({
+  handleTogglePdfSidebar() {
+    dispatch({
+      type: Constants.TOGGLE_PDF_SIDEBAR
+    });
+  }
+});
+const mapSidebarStateToProps = (state) => {
+  return {
```
Comment (NickHeiner) [inline client/app/components/PdfSidebar.jsx]: Please use the shorthand syntax when there's only a single expression in the function.

File: client/app/components/PdfSidebar.jsx [hunk 3]
```diff
@@ -178,5 +196,7 @@ PdfSidebar.propTypes = {
   onCancelCommentEdit: PropTypes.func,
   onCancelCommentAdd: PropTypes.func,
   onDeleteComment: PropTypes.func,
-  onJumpToComment: PropTypes.func
+  onJumpToComment: PropTypes.func,
```
Comment (NickHeiner) [inline client/app/components/PdfSidebar.jsx]: Reminder that the diff would be smaller and merging would be easier if you added new object entries anywhere but the end.

File: client/app/components/PdfUI.jsx [hunk 4]
```diff
@@ -103,16 +105,32 @@ export default class PdfUI extends React.Component {
...
+              </span>
+            </span>
+            <span className={categoryAndTypeGridSize}>
+              <Button
+                name="newTab"
+                classNames={['cf-pdf-button cf-pdf-cutoff cf-pdf-buttons-right']}
+                ariaLabel="open document in new tab"
+                onClick={() => window.open(
+                  linkToSingleDocumentView(this.props.doc), '_blank')}>
+                <span title={this.props.doc.type}>{this.props.doc.type}</span>
```
Comment (NickHeiner) [inline client/app/components/PdfUI.jsx]: Why do we have this `title` attribute?

File: client/app/reader/PdfViewer.jsx [hunk 5]
```diff
@@ -249,6 +250,17 @@ export default class PdfViewer extends React.Component {
...
+const mapStateToProps = (state) => {
+  return {
+    hidePdfSidebar: state.ui.pdf.hidePdfSidebar
+  };
+};
+
+export default connect(
+  mapStateToProps, null
+)(PdfViewer);
+
```
Comment (NickHeiner) [inline client/app/reader/PdfViewer.jsx]: nit: extra space

**Comment by NickHeiner** [comment]
This will fix the floating doc category icons issue.

```diff
diff --git a/app/assets/stylesheets/_reviewer.scss b/app/assets/stylesheets/_reviewer.scss
index 70cb5c37..4b3d52c7 100644
--- a/app/assets/stylesheets/_reviewer.scss
+++ b/app/assets/stylesheets/_reviewer.scss
@@ -131,6 +131,15 @@ svg[aria-hidden=true],
   z-index: 15;
   opacity: 1;
   padding: 15px 20px;
+
+  .category-icons-and-doc-type {
+    display: flex;
+    justify-content: flex-end;  
+
+    &>span:last-child {
+      max-width: 75%;
+    }
+  }
 }
 
 .cf-pdf-footer {
diff --git a/client/app/components/PdfUI.jsx b/client/app/components/PdfUI.jsx
index ff517958..23bc2208 100644
--- a/client/app/components/PdfUI.jsx
+++ b/client/app/components/PdfUI.jsx
@@ -104,35 +104,26 @@ export class PdfUI extends React.Component {
             <i className="fa fa-plus" aria-hidden="true"></i>
           </Button>
         </span>
-        <span className="usa-width-one-third">
-          <span className="usa-grid-full">
-            <span className={categoryAndTypeGridSize}>
-              <span className="cf-right-side">
-                <DocumentCategoryIcons docId={this.props.doc.id} />
-              </span>
-            </span>
-            <span className={categoryAndTypeGridSize}>
-              <Button
-                name="newTab"
-                classNames={['cf-pdf-button cf-pdf-cutoff cf-pdf-buttons-right']}
-                ariaLabel="open document in new tab"
-                onClick={() => window.open(
-                  linkToSingleDocumentView(this.props.doc), '_blank')}>
-                <span title={this.props.doc.type}>{this.props.doc.type}</span>
-              </Button>
-            </span>
-            {this.props.hidePdfSidebar && <span className="usa-width-one-third">
-              <Button
-                name="open menu"
-                classNames={['cf-pdf-button cf-pdf-cutoff']}
-                onClick={this.props.handleTogglePdfSidebar}>
-                <strong>
-                  Open menu
-                </strong>
-              </Button>
-            </span>}
-          </span>
+        <span className="usa-width-one-third category-icons-and-doc-type">
+          <DocumentCategoryIcons docId={this.props.doc.id} />
+          <Button
+            name="newTab"
+            classNames={['cf-pdf-button cf-pdf-cutoff cf-pdf-buttons-right']}
+            ariaLabel="open document in new tab"
+            onClick={() => window.open(
+              linkToSingleDocumentView(this.props.doc), '_blank')}>
+            <span title={this.props.doc.type}>{this.props.doc.type}</span>
+          </Button>
         </span>
+        {this.props.hidePdfSidebar &&
+          <Button
+            name="open menu"
+            classNames={['cf-pdf-button cf-pdf-cutoff']}
+            onClick={this.props.handleTogglePdfSidebar}>
+            <strong>
+              Open menu
+            </strong>
+          </Button>}
       </div>
       <div className="cf-pdf-navigation">
         { this.props.onPreviousPdf &&

```

File: client/app/components/PdfUI.jsx (hunk 4 above)
Comment (mdbenjam) [inline client/app/components/PdfUI.jsx]: It puts the full name as a tooltip. We'll probably want something more robust later. Another thing that Gina and I discussed offline. Putting it in the AC.

**Comment by NickHeiner** [comment]
Reminder to fix vertical spacing:

image

**Comment by NickHeiner** [comment]
👍 padding is fixed

**Comment by NickHeiner** [comment]
```diff
diff --git a/app/assets/stylesheets/react/_comment.scss b/app/assets/stylesheets/react/_comment.scss
index 07b28151..3b0fdd58 100644
--- a/app/assets/stylesheets/react/_comment.scss
+++ b/app/assets/stylesheets/react/_comment.scss
@@ -33,7 +33,6 @@
   padding: 15px;
   margin-bottom: 15px;
   border: 1px solid $color-gray-lighter;
-  max-width: 625px;
 }
 
 .comment-horizontal-container {

```

**Pull request merged**